Client/mods/deathmatch/logic/CClientGame.cpp

```diff
@@ -236,6 +236,8 @@ CClientGame::CClientGame(bool bLocalPlay)
     // Singular file download manager
     m_pSingularFileDownloadManager = new CSingularFileDownloadManager();
 
+    m_pPrimitiveBufferManager = new CClientPrimitiveBufferManager(m_pManager);
```

Comment:
Where are you deallocating `m_pPrimitiveBufferManager`?

Reply:
i think i doesn't deallocate it

Reply:
It's causing a memory leak. It's better to use `std::unique_ptr` here. It's a smart pointer, it will automatically get deleted when it goes out of scope, i.e., when client.dll gets unloaded. 

```
static std::unique_ptr<CClientPrimitiveBufferManager> m_pPrimitiveBufferManager= nullptr;

// In CClientGame::CClientGame
m_pPrimitiveBufferManager = std::make_unique<CClientPrimitiveBufferManager>(m_pManager);
```